File: output/0102.二叉树的层序遍历/0102-二叉树的层序遍历.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        ans = deque([root])
        res = []
        def bfs():
            while ans:
                temp = []
                for _ in range(len(ans)):
                    now = ans.popleft()
                    temp.append(now.val)
                    if now.left:
                        ans.append(now.left)
                    if now.right:
                        ans.append(now.right)
                res.append(temp)
        bfs()
        return res

File: 答题/标签/二叉树/0102.二叉树的层序遍历/0102-二叉树的层序遍历.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {
	res := [][]int{}
	var dfs func(root *TreeNode, index int)
	dfs = func(root *TreeNode, index int) {
		if root == nil {
			return
		}
		if index > len(res) {
			res = append(res, []int{root.Val})
		} else {
			res[index-1] = append(res[index-1], root.Val)
		}
		dfs(root.Left, index+1)
		dfs(root.Right, index+1)
	}
	dfs(root, 1)
	return res
}